core/src/test/java/com/sap/ai/sdk/core/AiCoreServiceTest.java

```diff
@@ -63,17 +68,30 @@ void testCustomBaseDestination() {
 
   @Test
   void testGetInferenceDestination() {
-    var builder = service.getInferenceDestination();
+    // default resource group destination
+    val builder = service.getInferenceDestination();
     assertThatThrownBy(() -> builder.forScenario("doesn't exist"))
         .isExactlyInstanceOf(DeploymentResolutionException.class);
 
+    // custom resource group destination
     val destination = service.getInferenceDestination("foo").usingDeploymentId("123");
-
     assertThat(destination.getUri())
         .hasHost("api.ai.com")
         .hasPath("/v2/inference/deployments/123/");
-
     assertThat(destination.getHeaders()).containsExactly(new Header("AI-Resource-Group", "foo"));
+
+    // scenario-based destination
+    val d = "{\"count\":1,\"resources\":[{\"id\":\"0123456789abcdef\",\"scenarioId\":\"foobar\"}]}";
+    val server = new WireMockServer(wireMockConfig().dynamicPort());
+    server.start();
+    server.stubFor(get(urlEqualTo("/v2/lm/deployments")).willReturn(okJson(d)));
+    HttpDestination foobar =
+        service
+            .withBaseDestination(DefaultHttpDestination.builder(server.baseUrl() + "/v2/").build())
+            .getInferenceDestination()
+            .forScenario("foobar");
+    assertThat(foobar.getUri()).hasPath("/v2/inference/deployments/0123456789abcdef/");
+    server.stop();
```

Comment:
(Question/Minor)

This looks like a drive-by(?). Or is it related to your PR? If not, I would rather move this out to a dedicated PR.